aikau/src/test/resources/alfresco/forms/controls/BaseFormTest.js

```diff
@@ -29,146 +29,193 @@ define(["intern!object",
+                  assert.lengthOf(elements, 2, "OK/Cancel buttons not found on basic form");
+               })
+               .end()
+
+            .findAllByCssSelector("#AUTOSAVE_FORM .confirmationButton, #AUTOSAVE_FORM .cancelButton")
+               .then(function(elements) {
+                  assert.lengthOf(elements, 0, "OK/Cancel buttons found on autosave form");
+               });
+         },
+
+         "Updating autosave value publishes form": function() {
+            return browser.findByCssSelector("#AUTOSAVE_FORM_FIELD .dijitInputInner")
+               .clearValue()
+               .type("wibble")
+               .getLastPublish("AUTOSAVE_FORM_1")
+               .then(function(payload) {
+                  assert.propertyVal(payload, "control", "wibble", "Did not autosave updated value");
+                  assert.propertyVal(payload, "alfValidForm", true, "Did not autosave validity property");
+               });
+         },
+
+         "Updating to invalid value does not autosave form": function() {
+            return browser.findById("CLEAR_AUTOSAVE_1")
+               .click()
+               .clearLog()
+               .getAllPublishes("AUTOSAVE_FORM_1")
+               .then(function(payloads) {
+                  assert.lengthOf(payloads, 0, "Published form when invalid");
+               });
+         },
+
+         "Autosave on invalid flag publishes invalid form": function() {
+            return browser.findByCssSelector("#AUTOSAVE_INVALID_FORM_FIELD .dijitInputInner")
+               .clearLog()
+               .clearValue()
+               .pressKeys(keys.BACKSPACE) // Need to trigger an update!
+               .getLastPublish("AUTOSAVE_FORM_2")
+               .then(function(payload) {
+                  assert.propertyVal(payload, "control", "", "Did not autosave updated, invalid value");
+                  assert.propertyVal(payload, "alfValidForm", false, "Did not autosave validity property");
+               });
+         },
+
+         "Autosaving with defined payload mixes payload into form values": function(){
+            return browser.findByCssSelector("#AUTOSAVE_FORM_FIELD") // Need to get session to check for publish
+               .getLastPublish("AUTOSAVE_FORM_2")
+               .then(function(payload) {
+                  assert.propertyVal(payload, "customProperty", "awooga", "Did not mix custom payload into form values");
+               });
+         },
+
+         "Post Coverage Results": function() {
+            TestCommon.alfPostCoverageResults(this, browser);
+         }
+      };
+   });
+
+   // This test reloads the page to clear any previous focus
+   registerSuite(function(){
+      var browser;
+
+      return {
+         name: "Base Form Control Tests - Focus and Enter to Submit",
+
+         setup: function() {
+            browser = this.remote;
+            return TestCommon.loadTestWebScript(this.remote, "/BaseForm", "Base Form Control Tests").end();
```

Comment:
I would have preferred an individual test page and test suite for this capability (see some of the other form tests) this helps when browsing the available pages from the unit test index page.